File: Counting/1493. Longest Subarray of 1's After Deleting One Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        nums.append(0)
        ans, count, prev, n = 0, 0, 0, len(nums)
        for val in nums:
            count += val
            if not val:
                ans = max(ans, count + prev)
                prev, count = count, 0            
        ans -= ans == n - 1
        return ans

File: 1493. Longest Subarray of 1's After Deleting One Element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) 
    {
        int ans = 0, n = nums.size();
        int i = 0, j = 0, zero_count = 0;
        while(j < n)
        {
            zero_count += nums[j++] == 0;
            while(zero_count > 1) 
                zero_count -= nums[i++] == 0;
            ans = max(ans, j - i - zero_count);
        }
        ans -= ans == n;
        return ans;
    }
};